catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/VisualPlacementActivityInputTest.java

```diff
@@ -0,0 +1,100 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.uiespresso.content.brick.app;
+
+import android.support.test.espresso.intent.Intents;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.bricks.PlaceAtBrick;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.catrobat.catroid.ui.SpriteActivity;
+import org.catrobat.catroid.uiespresso.util.rules.FragmentActivityTestRule;
+import org.catrobat.catroid.visualplacement.VisualPlacementActivity;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.intent.Intents.intended;
+import static android.support.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static android.support.test.espresso.intent.matcher.IntentMatchers.hasExtra;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.content.bricks.Brick.BrickField.X_POSITION;
+import static org.catrobat.catroid.ui.SpriteActivity.EXTRA_X_TRANSFORM;
+import static org.catrobat.catroid.ui.SpriteActivity.EXTRA_Y_TRANSFORM;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils.createProjectAndGetStartScript;
+import static org.hamcrest.core.AllOf.allOf;
+
+@RunWith(AndroidJUnit4.class)
+public class VisualPlacementActivityInputTest {
+	private static final int XPOS = -200;
+	private static final int YPOS = 500;
+	private static final String INVALID_TEST_STRING = "test";
+	private PlaceAtBrick bricktoAdd;
+
+	public FragmentActivityTestRule<SpriteActivity> baseActivityTestRule = new
+			FragmentActivityTestRule<>(SpriteActivity.class, SpriteActivity.EXTRA_FRAGMENT_POSITION,
+			SpriteActivity.FRAGMENT_SCRIPTS);
+
+	@Before
+	public void setUp() throws Exception {
+		bricktoAdd = new PlaceAtBrick();
+		createProjectAndGetStartScript(VisualPlacementActivityInputTest.class.getSimpleName())
+				.addBrick(bricktoAdd);
+		Intents.init();
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		Intents.release();
+		TestUtils.deleteProjects(PlaceAtBrickTest.class.getSimpleName());
+	}
+
+	@Test
+	public void testCorrectInput() {
+		bricktoAdd.setCoordinates(XPOS, YPOS);
+		baseActivityTestRule.launchActivity();
+		onView(withId(R.id.brick_place_at_edit_text_x)).perform(click());
+		onView(withText(R.string.brick_place_at_option_place_visually)).perform(click());
```

Comment:
this will still launch the VisualPlacementActivity though.
the test is over at this point, but you leave the environment in a different activity than what youre actually testing.

theres two ways around this:
intending(intentMatcher).respondWith(result);

or 

		Instrumentation.ActivityMonitor activityMonitor = new Instrumentation.ActivityMonitor
				(VisualPlacementActivity.class.getName(), null, true);
		InstrumentationRegistry.getInstrumentation().addMonitor(activityMonitor);
		assertTrue(InstrumentationRegistry.getInstrumentation().checkMonitorHit(activityMonitor, 1));


both prevent the VisualPlacementActivity from being launched, while asserting that it would be and instead return a activityResult right away.

that also points you towards the untested parts of your PR.

you got:
- VisualPlacementActivity returns a valid activityResult if it is started with this very coordinates
- ScriptActivity starts VisualPlacementActivity with a valid intent containing these coordinates

whats missing: (untested)
- ScriptActivity puts the returned value in the brick and uses them
- VisualPlacement activity returning something different than what it got in its start intent.

this can be easily tested in the scope of those two tests. please move them to the intents package, and name them accordingly to the activity they test.

VisualPlacementActivity:
check that incoming Intent sets the x and y coords in VisualPlacementActivity
and check that outcoing Intent sends the modified x and y coords back (theres already setter methods in this activity)

SpriteActivity
check that it launches an intent for the VisualPlacementActivity with the given formula field values  (already implemented)
and check that SpriteActivity handles the activityResult and does something with those values (missing)